### PR TITLE
ETFE-562 added text area component

### PR DESCRIPTION
**Pull request by david-walters-HMRC**

### CODE REVIEW

File: .g8/characterCountPage/app/views/$className$View.scala.html [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+govukErrorSummary: GovukErrorSummary,
+govukCharacterCount : GovukCharacterCount,
+govukButton: GovukButton,
+h1: components.h1,
+saveAndContinue: components.saveAndContinue
+)
+
+@(form: Form[_], mode: Mode)(implicit request: DataRequest[_], messages: Messages)
+
+@layout(pageTitle = title(form, messages("moreInformation.title"))) {
```
Comment (MJCallahanPage) [inline .g8/characterCountPage/app/views/$className$View.scala.html:17]:
```suggestion
@layout(pageTitle = title(form, messages("$className;format="decap"$.title"))) {
```

File: .g8/characterCountPage/app/views/$className$View.scala.html [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+@formHelper(action = routes.MoreInformationController.onSubmit(request.ern, request.arc, mode)) {
```
Comment (MJCallahanPage) [inline .g8/characterCountPage/app/views/$className$View.scala.html:19]:
```suggestion
@formHelper(action = routes.$className$Controller.onSubmit(request.ern, request.arc, mode)) {
```

File: .g8/characterCountPage/app/views/$className$View.scala.html [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+@if(form.errors.nonEmpty) {
+@govukErrorSummary(ErrorSummaryViewModel(form))
+}
+
+@{govukCharacterCount(CharacterCount(
+id = "more-information",
+name = "more-information",
```
Comment (MJCallahanPage) [inline .g8/characterCountPage/app/views/$className$View.scala.html:27]:
```suggestion
id = "value",
name = "value",
```

File: .g8/characterCountPage/app/views/$className$View.scala.html [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+maxLength = Some(350),
+label = LabelViewModel(HtmlContent(heading)).asPageHeading(LabelSize.Large),
+value = form("more-information").value,
```
Comment (MJCallahanPage) [inline .g8/characterCountPage/app/views/$className$View.scala.html:30]:
```suggestion
value = form("value").value,
```

File: .g8/characterCountPage/app/views/$className$View.scala.html [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+errorMessage = form.errors("more-information") match {
```
Comment (MJCallahanPage) [inline .g8/characterCountPage/app/views/$className$View.scala.html:31]:
```suggestion
errorMessage = form.errors("value") match {
```

File: .g8/characterCountPage/app/views/$className$View.scala.html [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+)}
+
+@saveAndContinue()
+
+}
+}
+
+@heading = {
+<span class="govuk-caption-l">@messages("arc.subHeading", request.arc)</span>
+@messages("moreInformation.heading")
```
Comment (MJCallahanPage) [inline .g8/characterCountPage/app/views/$className$View.scala.html:45]:
```suggestion
@messages("$className;format="decap"$.heading")
```

File: .g8/characterCountPage/default.properties [hunk 7]
```diff
@@ -0,0 +1,3 @@
+description = Generates a controller and view for a page with a single string question
```
Comment (MJCallahanPage) [inline .g8/characterCountPage/default.properties:1]:
```suggestion
description = Generates a controller and view for a page with a CharacterCounter question
```

File: .g8/characterCountPage/default.properties [hunk 8]
```diff
@@ -0,0 +1,3 @@
...
+className = StringPageName
```
Comment (MJCallahanPage) [inline .g8/characterCountPage/default.properties:2]:
```suggestion
className = CharacterCounterPageName
```

File: .g8/characterCountPage/app/views/$className$View.scala.html [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+@formHelper(action = routes.MoreInformationController.onSubmit(request.ern, request.arc, mode)) {
+
+@if(form.errors.nonEmpty) {
+@govukErrorSummary(ErrorSummaryViewModel(form))
+}
+
+@{govukCharacterCount(CharacterCount(
+id = "more-information",
+name = "more-information",
+maxLength = Some(350),
```
Comment (MJCallahanPage) [inline .g8/characterCountPage/app/views/$className$View.scala.html:28]:
```suggestion
maxLength = Some($maxLength$),
```

**Review by MJCallahanPage — DISMISSED** [review]
Some small tweaks, mainly in the view. To use the `g8` template parameters for the naming of fields/classes.

**Review by stephenkward — APPROVED** [review]

**Pull request merged**